### PR TITLE
Improve APT program generation interface.

### PR DESCRIPTION
**Pull request by schlafly**
This PR adds a handful of things to better support generating APT programs with romanisim.

* It removes the old apt interface which we have never really used.
* It changes the filenames to use the new scheme that omits the group, sequence, and activity bits.
* It reads different metadata out of the APT file, aiming to be able to fill out PI / category / subcategory / ... information, but it doesn't yet pass that information in a sensible way forward when spitting out a script file.
* It adds a new merge_dicts utility function that does a recursive dictionary merger, so that we can update keys from APT into the main metadata dictionary.
* It adds a new script to generate Gaia catalogs for a region.
* It tweaks romanisim-make-stack somewhat to use the new APT output format (which includes plan/pass/segment/observation/visit information) and the new metadata grabber from APT.

### CODE REVIEW

File: romanisim/util.py [hunk 1]
```diff
@@ -544,3 +544,31 @@ def update_photom_keywords(im, gain=None):
         0 * u.MJy / u.sr)
     im['meta']['photometry']['conversion_microjanskys_uncertainty'] = (
         0 * u.uJy / u.arcsec ** 2)
+
+
+def merge_dicts(a, b):
```
Comment (PaulHuwe) [inline romanisim/util.py:549]: Should we update the code to use this in other places?

File: scripts/romanisim-make-stack [hunk 2]
```diff
@@ -92,8 +85,8 @@ def main():
     # SCA detector identifier number
     parser.add_argument('-s', '--sca',
                         type=int,
-                        default=0,
-                        help='SCA to simulate')
+                        default=-1,
```
Comment (PaulHuwe) [inline scripts/romanisim-make-stack:88]: Did you ensure that this behaved as expected for both a specified SCA and for all?

**Review by PaulHuwe — APPROVED** [review]
We should add tests of parse_apt_file, ramanisim-gaia-catalog, romanisim-make-catalog, romanisim-make-image, and romanisim-make-stack.

File: romanisim/util.py (hunk 1 above)
Comment (schlafly) [reply romanisim/util.py:549]: We should, but I haven't done that yet.

File: scripts/romanisim-make-stack (hunk 2 above)
Comment (schlafly) [reply scripts/romanisim-make-stack:88]: I only tested the "commands" mode, but this works for both single and multiple SCAs at present.

**Pull request merged**